kmath-dimensions/src/commonMain/kotlin/space/kscience/kmath/dimensions/Wrappers.kt

```diff
@@ -23,11 +24,11 @@ public interface DMatrix<T, R : Dimension, C : Dimension> : Structure2D<T> {
          * Coerces a regular matrix to a matrix with type-safe dimensions and throws a error if coercion failed
          */
         public inline fun <T, reified R : Dimension, reified C : Dimension> coerce(structure: Structure2D<T>): DMatrix<T, R, C> {
-            require(structure.rowNum == Dimension.dim<R>().toInt()) {
+            require(structure.rowNum == Dimension.dim<R>().toIntExact()) {
```

Comment:
Maybe it is better to use UInt for rowNumbers?

Reply:
`Structure2D.rowNum: UInt`?

Reply:
yes. It does not make sense to use Ints now that UInt is stable

Reply:
Should I try to do it in this branch?

Reply:
Yes, I think so